src/components/wasteCalendar/ReminderSettingsReducer.ts

```diff
@@ -43,7 +43,7 @@ export const reminderSettingsReducer: React.Reducer<ReminderSettings, ReminderSe
     case ReminderSettingsActionType.UPDATE_ACTIVE_TYPE: {
       const { key, value } = action.payload;
       const newState = { ...state };
-      newState.activeTypes[key] = value;
+      newState.activeTypes[key] = { ...newState.activeTypes[key], active: value };
```

Comment:
can `newState.activeTypes` be null or undefined, so that `[key]` would break?

Reply:
it can not unless we do not initialize it when we call `useReducer`. and that will cause a type error, if we try to pass in a faulty initial value